### PR TITLE
WFLY-17121 Remove preview-feature-pack-common-[microprofile-]licenses…

### PR DESCRIPTION
**Pull request by RanabirChakraborty**
Issue: https://issues.redhat.com/browse/WFLY-17121

### CODE REVIEW

File: ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml [hunk 1]
```diff
@@ -24,8 +24,63 @@
...
+                <license>
+                    <name>GNU Lesser General Public License v2.1 or later</name>
+                    <url>http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html</url>
+                    <distribution>repo</distribution>
+                </license>
+            </licenses>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>wildfly-standard-expansion-bom</artifactId>
```
Comment (bstansberry) [inline ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml:83]: These boms aren't actually dependencies and are not included in a WildFly Preview installation for which the licenses files that come from these is produced. So these can be removed and this file can not have any dependency entries at all.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml [hunk 2]
```diff
@@ -2,545 +2,19 @@
...
-      <groupId>${project.groupId}</groupId>
-      <artifactId>wildfly-ee-9-deployment-transformer</artifactId>
-      <licenses>
-        <license>
-          <name>GNU Lesser General Public License v2.1 or later</name>
-          <url>http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html</url>
-          <distribution>repo</distribution>
-        </license>
-      </licenses>
-    </dependency>
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:14]: This entry should be kept.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml [hunk 3]
```diff
@@ -551,172 +25,7 @@
...
-    <dependency>
-      <groupId>${project.groupId}</groupId>
-      <artifactId>wildfly-undertow</artifactId>
+      <artifactId>wildfly-common-ee-dependency-management</artifactId>
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:28]: We shouldn't have entries for the boms.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml (hunk 2 above)
Comment (bstansberry) [reply ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:14]: There are a number of dependency entries in https://github.com/wildfly/wildfly/blob/main/ee-9/common/pom.xml that now don't have matching entries in this file. They all should.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml (hunk 3 above)
Comment (RanabirChakraborty) [reply ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml]: @bstansberry Changed everything accordingly, can you please review it once.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml [hunk 4]
```diff
@@ -2,677 +2,8 @@
...
-          <url>http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html</url>
-          <distribution>repo</distribution>
-        </license>
-      </licenses>
-    </dependency>
-    <dependency>
-      <groupId>${project.groupId}</groupId>
-      <artifactId>wildfly-transactions</artifactId>
+      <groupId>org.hibernate.search</groupId>
+      <artifactId>hibernate-search-mapper-orm-coordination-outbox-polling-orm6</artifactId>
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:6]: This entry should be moved down to be in alphabetical order.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml [hunk 5]
```diff
@@ -682,74 +13,74 @@
       </licenses>
     </dependency>
     <dependency>
-      <groupId>${project.groupId}</groupId>
-      <artifactId>wildfly-rts</artifactId>
+      <groupId>org.latencyutils</groupId>
+      <artifactId>LatencyUtils</artifactId>
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:17]: This entry should be moved down to be in alphabetical order.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml [hunk 6]
```diff
@@ -682,74 +13,74 @@
...
+          <name>Apache License 2.0</name>
+          <url>http://www.apache.org/licenses/LICENSE-2.0</url>
           <distribution>repo</distribution>
         </license>
       </licenses>
     </dependency>
     <dependency>
-      <groupId>${project.groupId}</groupId>
-      <artifactId>wildfly-undertow</artifactId>
+      <groupId>io.micrometer</groupId>
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:49]: The groupId here should be io.prometheus.

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml [hunk 7]
```diff
@@ -682,74 +13,74 @@
...
+          <name>Apache License 2.0</name>
+          <url>http://www.apache.org/licenses/LICENSE-2.0</url>
           <distribution>repo</distribution>
         </license>
       </licenses>
     </dependency>
     <dependency>
-      <groupId>${project.groupId}</groupId>
-      <artifactId>wildfly-weld-bean-validation</artifactId>
+      <groupId>io.micrometer</groupId>
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:60]: The groupId here should be io.prometheus.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Hi @RanabirChakraborty -- I had a chance to review again and I have some more comments.

File: ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml [hunk 8]
```diff
@@ -2,8 +2,8 @@
 <licenseSummary>
     <dependencies>
         <dependency>
-            <groupId>${full.maven.groupId}</groupId>
-            <artifactId>wildfly-microprofile-jwt-smallrye</artifactId>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>wildfly-preview-ee-bom</artifactId>
```
[the remark this answers is not in the document]
Comment (RanabirChakraborty) [reply ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml:6]: If I removed these two dependencies from `preview-feature-pack-common-microprofile-licenses.xml` there will be no dependencies left.

File: ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml [hunk 9]
```diff
@@ -13,19 +13,8 @@
...
-                    <url>http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html</url>
-                    <distribution>repo</distribution>
-                </license>
-            </licenses>
-        </dependency>
-        <dependency>
-            <groupId>${full.maven.groupId}</groupId>
-            <artifactId>wildfly-microprofile-health-smallrye</artifactId>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>wildfly-preview-expansion-bom</artifactId>
```
[the remark this answers is not in the document]
Comment (RanabirChakraborty) [reply ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml:17]: Same here

File: ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml (hunk 4 above)
Comment (RanabirChakraborty) [reply ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml:6]: I guess it's in alphabetical order only by `artifactId` name, Like -> h> L> m> r> s> w

**Comment by bstansberry** [comment]
Thanks @RanabirChakraborty -- these files should be correct in main now.

**Comment by RanabirChakraborty** [comment]
Thanks @bstansberry

**Pull request closed without merging**